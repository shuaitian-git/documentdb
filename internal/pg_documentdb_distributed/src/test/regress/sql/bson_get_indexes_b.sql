SET search_path TO documentdb_core,documentdb_api,documentdb_api_catalog,documentdb_api_internal;
SET citus.next_shard_id TO 2200000;
SET documentdb.next_collection_id TO 22000;
SET documentdb.next_collection_index_id TO 22000;

-- --Test 1 Compound index test --
SELECT documentdb_api_internal.create_indexes_non_concurrently('orderdb', documentdb_distributed_test_helpers.generate_create_index_arg('compound_orders', 'comp_index', '{"total_price": 1,"shipping_distance":1}'), true);
SELECT documentdb_api_internal.create_indexes_non_concurrently('orderdb', documentdb_distributed_test_helpers.generate_create_index_arg('compound_orders', 'comp_index1', '{"delivery_hours": 1,"customer_rating":1}'), true);
select documentdb_api.list_indexes_cursor_first_page('orderdb', '{ "listIndexes": "compound_orders" }') ORDER BY 1;

--Test 2 Descending Index --
SELECT documentdb_api_internal.create_indexes_non_concurrently('orderdb', documentdb_distributed_test_helpers.generate_create_index_arg('order_items', 'test1', '{"item_count": -1}'), true);
SELECT documentdb_api_internal.create_indexes_non_concurrently('orderdb', documentdb_distributed_test_helpers.generate_create_index_arg('order_items', 'test2', '{"to_city": -1}'), true);
select documentdb_api.list_indexes_cursor_first_page('orderdb', '{ "listIndexes": "order_items" }') ORDER BY 1;

--Test 3 Ascending  Descending Index --
SELECT documentdb_api_internal.create_indexes_non_concurrently('orderdb', documentdb_distributed_test_helpers.generate_create_index_arg('order_shipments', 'test1', '{"promo_code": 1}'), true);
SELECT documentdb_api_internal.create_indexes_non_concurrently('orderdb', documentdb_distributed_test_helpers.generate_create_index_arg('order_shipments', 'test2', '{"order_status": -1}'), true);
select documentdb_api.list_indexes_cursor_first_page('orderdb', '{ "listIndexes": "order_shipments" }') ORDER BY 1;
