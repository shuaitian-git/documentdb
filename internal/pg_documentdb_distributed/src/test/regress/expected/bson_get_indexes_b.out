SET search_path TO documentdb_core,documentdb_api,documentdb_api_catalog,documentdb_api_internal;
SET citus.next_shard_id TO 2200000;
SET documentdb.next_collection_id TO 22000;
SET documentdb.next_collection_index_id TO 22000;
-- --Test 1 Compound index test --
SELECT documentdb_api_internal.create_indexes_non_concurrently('orderdb', documentdb_distributed_test_helpers.generate_create_index_arg('compound_orders', 'comp_index', '{"total_price": 1,"shipping_distance":1}'), true);
NOTICE:  creating collection
                                                                                                   create_indexes_non_concurrently                                                                                                   
---------------------------------------------------------------------
 { "raw" : { "defaultShard" : { "numIndexesBefore" : { "$numberInt" : "1" }, "numIndexesAfter" : { "$numberInt" : "2" }, "createdCollectionAutomatically" : true, "ok" : { "$numberInt" : "1" } } }, "ok" : { "$numberInt" : "1" } }
(1 row)

SELECT documentdb_api_internal.create_indexes_non_concurrently('orderdb', documentdb_distributed_test_helpers.generate_create_index_arg('compound_orders', 'comp_index1', '{"delivery_hours": 1,"customer_rating":1}'), true);
                                                                                                   create_indexes_non_concurrently                                                                                                    
---------------------------------------------------------------------
 { "raw" : { "defaultShard" : { "numIndexesBefore" : { "$numberInt" : "2" }, "numIndexesAfter" : { "$numberInt" : "3" }, "createdCollectionAutomatically" : false, "ok" : { "$numberInt" : "1" } } }, "ok" : { "$numberInt" : "1" } }
(1 row)

select documentdb_api.list_indexes_cursor_first_page('orderdb', '{ "listIndexes": "compound_orders" }') ORDER BY 1;
                                                                                                                                                                                                                                                                                                                  list_indexes_cursor_first_page                                                                                                                                                                                                                                                                                                                  
---------------------------------------------------------------------
 ("{ ""cursor"" : { ""id"" : { ""$numberLong"" : ""0"" }, ""ns"" : ""orderdb.compound_orders"", ""firstBatch"" : [ { ""v"" : { ""$numberInt"" : ""2"" }, ""key"" : { ""_id"" : { ""$numberInt"" : ""1"" } }, ""name"" : ""_id_"" }, { ""v"" : { ""$numberInt"" : ""2"" }, ""key"" : { ""total_price"" : { ""$numberInt"" : ""1"" }, ""shipping_distance"" : { ""$numberInt"" : ""1"" } }, ""name"" : ""comp_index"" }, { ""v"" : { ""$numberInt"" : ""2"" }, ""key"" : { ""delivery_hours"" : { ""$numberInt"" : ""1"" }, ""customer_rating"" : { ""$numberInt"" : ""1"" } }, ""name"" : ""comp_index1"" } ] }, ""ok"" : { ""$numberDouble"" : ""1.0"" } }",,f,0)
(1 row)

--Test 2 Descending Index --
SELECT documentdb_api_internal.create_indexes_non_concurrently('orderdb', documentdb_distributed_test_helpers.generate_create_index_arg('order_items', 'test1', '{"item_count": -1}'), true);
NOTICE:  creating collection
                                                                                                   create_indexes_non_concurrently                                                                                                   
---------------------------------------------------------------------
 { "raw" : { "defaultShard" : { "numIndexesBefore" : { "$numberInt" : "1" }, "numIndexesAfter" : { "$numberInt" : "2" }, "createdCollectionAutomatically" : true, "ok" : { "$numberInt" : "1" } } }, "ok" : { "$numberInt" : "1" } }
(1 row)

SELECT documentdb_api_internal.create_indexes_non_concurrently('orderdb', documentdb_distributed_test_helpers.generate_create_index_arg('order_items', 'test2', '{"to_city": -1}'), true);
                                                                                                   create_indexes_non_concurrently                                                                                                    
---------------------------------------------------------------------
 { "raw" : { "defaultShard" : { "numIndexesBefore" : { "$numberInt" : "2" }, "numIndexesAfter" : { "$numberInt" : "3" }, "createdCollectionAutomatically" : false, "ok" : { "$numberInt" : "1" } } }, "ok" : { "$numberInt" : "1" } }
(1 row)

select documentdb_api.list_indexes_cursor_first_page('orderdb', '{ "listIndexes": "order_items" }') ORDER BY 1;
                                                                                                                                                                                                                                                    list_indexes_cursor_first_page                                                                                                                                                                                                                                                     
---------------------------------------------------------------------
 ("{ ""cursor"" : { ""id"" : { ""$numberLong"" : ""0"" }, ""ns"" : ""orderdb.order_items"", ""firstBatch"" : [ { ""v"" : { ""$numberInt"" : ""2"" }, ""key"" : { ""_id"" : { ""$numberInt"" : ""1"" } }, ""name"" : ""_id_"" }, { ""v"" : { ""$numberInt"" : ""2"" }, ""key"" : { ""item_count"" : { ""$numberInt"" : ""-1"" } }, ""name"" : ""test1"" }, { ""v"" : { ""$numberInt"" : ""2"" }, ""key"" : { ""to_city"" : { ""$numberInt"" : ""-1"" } }, ""name"" : ""test2"" } ] }, ""ok"" : { ""$numberDouble"" : ""1.0"" } }",,f,0)
(1 row)

--Test 3 Ascending  Descending Index --
SELECT documentdb_api_internal.create_indexes_non_concurrently('orderdb', documentdb_distributed_test_helpers.generate_create_index_arg('order_shipments', 'test1', '{"promo_code": 1}'), true);
NOTICE:  creating collection
                                                                                                   create_indexes_non_concurrently                                                                                                   
---------------------------------------------------------------------
 { "raw" : { "defaultShard" : { "numIndexesBefore" : { "$numberInt" : "1" }, "numIndexesAfter" : { "$numberInt" : "2" }, "createdCollectionAutomatically" : true, "ok" : { "$numberInt" : "1" } } }, "ok" : { "$numberInt" : "1" } }
(1 row)

SELECT documentdb_api_internal.create_indexes_non_concurrently('orderdb', documentdb_distributed_test_helpers.generate_create_index_arg('order_shipments', 'test2', '{"order_status": -1}'), true);
                                                                                                   create_indexes_non_concurrently                                                                                                    
---------------------------------------------------------------------
 { "raw" : { "defaultShard" : { "numIndexesBefore" : { "$numberInt" : "2" }, "numIndexesAfter" : { "$numberInt" : "3" }, "createdCollectionAutomatically" : false, "ok" : { "$numberInt" : "1" } } }, "ok" : { "$numberInt" : "1" } }
(1 row)

select documentdb_api.list_indexes_cursor_first_page('orderdb', '{ "listIndexes": "order_shipments" }') ORDER BY 1;
                                                                                                                                                                                                                                                        list_indexes_cursor_first_page                                                                                                                                                                                                                                                         
---------------------------------------------------------------------
 ("{ ""cursor"" : { ""id"" : { ""$numberLong"" : ""0"" }, ""ns"" : ""orderdb.order_shipments"", ""firstBatch"" : [ { ""v"" : { ""$numberInt"" : ""2"" }, ""key"" : { ""_id"" : { ""$numberInt"" : ""1"" } }, ""name"" : ""_id_"" }, { ""v"" : { ""$numberInt"" : ""2"" }, ""key"" : { ""promo_code"" : { ""$numberInt"" : ""1"" } }, ""name"" : ""test1"" }, { ""v"" : { ""$numberInt"" : ""2"" }, ""key"" : { ""order_status"" : { ""$numberInt"" : ""-1"" } }, ""name"" : ""test2"" } ] }, ""ok"" : { ""$numberDouble"" : ""1.0"" } }",,f,0)
(1 row)

